core/src/main/java/org/elasticsearch/index/query/functionscore/FunctionScoreQueryBuilder.java

```diff
@@ -450,8 +459,34 @@ public FilterFunctionBuilder rewrite(QueryRewriteContext context) throws IOExcep
         return this;
     }
 
-    public static FunctionScoreQueryBuilder fromXContent(Function<String, ScoreFunctionParser<?>> scoreFunctionLookup,
-            QueryParseContext parseContext) throws IOException {
+    /**
+     * Looks up {@linkplain ScoreFunctionParser}s based on name and context. Needs context to check the ParseFieldMatcher and report nice
+     * errors.
+     */
+    public static class ScoreFunctionsRegistry {
```

Comment:
I'm pretty sure we broke that kind of thing a while back. I'll make that work and leave comments so it probably won't be broken again.